### ISSUE TITLE
Loading a ptycho object on a CPU machine, originally saved on a GPU-enabled machine

### ISSUE THREAD

**Issue by swelborn** [py4dstem/py4DSTEM#540]
**Describe the bug**
If you save a ptycho to an h5 that you reconstructed using a GPU, you cannot load it back in using a CPU machine.

**To Reproduce**
Save a ptycho with its flag set to 'gpu'. Try to open this on CPU-only machine. You get this:
```
Traceback (most recent call last):
  File "/analysis/scripts/plots.py", line 211, in <module>
    main()
  File "/analysis/scripts/plots.py", line 203, in main
    extracted_datas = load_multiple_datasets(base_path, min_scan_num, max_scan_num)
  File "/analysis/scripts/plots.py", line 107, in load_multiple_datasets
    extracted_datas.append(load_hdf5_data(scan_path))
  File "/analysis/scripts/plots.py", line 31, in load_hdf5_data
    ptycho = py4DSTEM.read(
  File "/usr/local/lib/python3.10/site-packages/py4DSTEM/io/read.py", line 95, in read
    data = emd.read(filepath, emdpath=datapath, tree=tree)
  File "/usr/local/lib/python3.10/site-packages/emdfile/read.py", line 229, in read
    node = _read_single_node(nodegroup)
  File "/usr/local/lib/python3.10/site-packages/emdfile/read.py", line 256, in _read_single_node
    data = __class__.from_h5(grp)
  File "/usr/local/lib/python3.10/site-packages/emdfile/classes/tree.py", line 572, in from_h5
    node = cls(**args)
  File "/usr/local/lib/python3.10/site-packages/py4DSTEM/process/phase/iterative_singleslice_ptychography.py", line 123, in __init__
    self._asnumpy = cp.asnumpy
AttributeError: 'NoneType' object has no attribute 'asnumpy'
```

**Expected behavior**
This should not be dependent on how it was originally saved.

**py4DSTEM version** 
14.3

**Python version**
3.10
**Operating system** 
both arm/amd64 show this.

**Comment by gvarnavi** [comment]
Ah shoot, this is a known bug Mohsen reported in the Slack channel and I promised a fix would make the next release, but forgot to include in latest PR..

Thanks for reporting, will try and include in the next `phase_contrast` updates!

**Comment by gvarnavi** [comment]
Actually, got around fixing it with commit 43220d0 and should be included in next release tomorrow when PR #539 is merged.

The fix isn't terribly elegant, but it's consistent with the current framework. Essentially all save calls should now set `device='cpu'` to be able to load on any machine. The user can then overwrite this post-loading (similar to how we save w/o a datacube and re-attach it later for space purposes). Something like this:

``` python
new_ptycho = py4DSTEM.read("../data/ptycho_test_save.h5")
new_ptycho= new_ptycho.attach_datacube(
    dataset
).reinitialize_parameters(
    device="gpu",
).preprocess(...).reconstruct(...)
```
